File: C++/Running_time_and_Complexity.cpp
#include <cmath>
#include <cstdio>
#include <vector>
#include <iostream>
#include <algorithm>
using namespace std;
void primechecker(int n){
    if(n==1){
        cout<<"Not prime"<<endl;
    }
    else if(n==2){
        cout<<"Prime"<<endl;
    }
    else{
        bool flag=false;
        if(n%2==0){
            flag=true;
        }
        else{
            int x=sqrt(n);
            for(int i=3;i<x+1;i+=2)
            {
                if(n%i==0){
                    flag=true;
}}}
        if(flag){
            cout<<"Not prime"<<endl;
        }
        else{
            cout<<"Prime"<<endl;
}}}

int main() {
    int T;
    cin>>T;
    while(T--){
        int temp;
        cin>>temp;
        primechecker(temp);
    }
    return 0;
}

File: Perl/Running_Time_and_Complexity.pl
use strict;
use warnings;
my $testcases=<STDIN>;
chomp $testcases;
for(my $i=0;$i<$testcases;$i=$i+1)
{
    my $n=<STDIN>;
    chomp $n;
    if($n==1)
    {
        print("Not prime\n");
    }
    elsif($n==2)
    {
        print("Prime\n");
    }
    elsif($n%2==0)
    {
        print("Not prime\n");
    }
    else
    {
        my $temp=sqrt($n);
        my $a=0;
        for(my $j=3;$j<=$temp+1;$j=$j+2)
        {
            if(($n%$j)==0)
            {
                $a=1;
            }
        }
        if($a==0)
        {
            print("Prime\n");
        }
        else
        {
            print("Not prime\n");
        }
    }
}
